### PR TITLE
Moved preview courses to background jobs

### PR DESCRIPTION
**Pull request by Dantemss**
Requires changes to tutor-deployment by @pandafulmanda (making different delayed_job workers process different queues)

### CODE REVIEW

File: config/initializers/delayed_job.rb [hunk 1]
```diff
@@ -15,8 +15,19 @@
...
+
+# Must be equal to the longest max_run_time below
+Delayed::Worker.max_run_time = 4.hours
+
+# Default queue priorities and max_run_times
+Delayed::Worker.queue_attributes = {
+  high_priority: { priority: -5, max_run_time: 1.minute   },
+  default:       { priority:  0, max_run_time: 5.minutes  },
+  low_priority:  { priority:  5, max_run_time: 30.minutes },
+  long_running:  { priority: 10, max_run_time: 4.hours    }
```
Comment (jpslav) [inline config/initializers/delayed_job.rb]: is 4 hours enough?

Comment (jpslav) [inline config/initializers/delayed_job.rb]: actually from below I'm not quite sure what the max_run_time is getting at.

File: config/initializers/delayed_job.rb [hunk 2]
```diff
@@ -92,6 +106,22 @@ def initialize_with_new_relic_fix(*args)
...
+      "WHEN '#{queue_name}' THEN #{attributes.fetch(:max_run_time, max_run_time)}"
+    end
+    max_run_time_sql = max_run_time_cases.empty? ?
+      max_run_time :
+      "CASE \"delayed_jobs\".\"queue\" #{max_run_time_cases.join(' ')} ELSE #{max_run_time} END"
+
+    where(
+      "(run_at <= ? AND (locked_at IS NULL OR locked_at + INTERVAL '1 second' * #{max_run_time_sql
+      } < ? ) OR locked_by = ?) AND failed_at IS NULL", db_time_now, db_time_now, worker_name
+    )
```
Comment (jpslav) [inline config/initializers/delayed_job.rb]: Probably could use a comment here to explain the intent

Comment (Dantemss) [inline config/initializers/delayed_job.rb]: OK, I'll add a comment.

File: config/initializers/delayed_job.rb (hunk 1 above)
Comment (Dantemss) [inline config/initializers/delayed_job.rb]: I think even the research export finishes in less than 4 hours right? At least for now...

**Comment by Dantemss** [comment]
@jpslav specs and comment added

File: config/initializers/delayed_job.rb (hunk 1 above)
Comment (Dantemss) [inline config/initializers/delayed_job.rb]: I'm actually considering removing these custom max_run_times and setting everything to 4 hours. The DJ timeout implementation is flawed and can cause data corruption (research export) or jobs running twice.

**Comment by Dantemss** [comment]
Removed per-queue max_run_times due to Delayed Job timeouts causing bad things to happen.

**Pull request merged**